### ISSUE TITLE
"manage-events" inconsistent behaviour when saving Admin Event config

### ISSUE THREAD

**Issue by mpalavrov** [keycloak/keycloak#35848]
### Before reporting an issue

- [X] I have read and understood the above terms for submitting issues, and I understand that my issue may be closed without action if I do not follow them.

### Area

admin/ui

### Describe the bug

When an user is having all roles, including "manage-realm", but not having "manage-events", and want to update the Admin Events config, an error is shown in the network tab and in the UI, but the config is actually updated

### Version

26.0.7

### Regression

- [ ] The issue is a regression

### Expected behavior

The Event Config to not be updated, as the "manage-events" role is not assigned to the user (or in case it is expected the user to be able to edit this setting, only by having "manage-realm", no error to be shown).

### Actual behavior

Error is displayed in the UI and in the Network Tab, but the config is updated anyway.

### How to Reproduce?

1. Create realm
2. Create user with password
3. Assign all "realm-management" roles, except "realm-admin" and "manage-events"
4. Open the security-console of the realm with the created user
5. Go to Realm settings - events - Admin events settings
6. Add new value for the "Expiration" for example
7. Save

![2024-07-31_16h03_40](https://github.com/user-attachments/assets/a4e2a6c6-7cd9-4509-a66e-ad78cc53909f)

**Comment by ssilvert** [comment]
I have verified this bug.  

The reason this happens is because in this case the user is allowed to call the REST API that saves the realm, i.e. `PUT http://localhost:8180/admin/realms/test`

However, this user is not allowed to call the the API that saves the events config, `PUT http://localhost:8180/admin/realms/test/events/config`.  That is what returns the 403.

Because some of the `Admin events settings` are saved as realm attributes, the UI is required to call both when saving these settings.  So one of the calls succeeds and the other fails.

The core team will need to fix this on the back end.  IMO, none of the events config should be saved as realm attributes.  If that change is made then we will need to also change the UI.

Changing ownership to `core/iam`.

**Comment by mposolda** [comment]
@pedroigor I agree that we can go with (2) as a bugfix. It looks that during realm upgrade admin request, we can compare sent realm with the realm from the DB. If events related attributes are updated and user doesn't have `manage-events`, we can throw authorization error?

**Comment by pedroigor** [comment]
The only problem I see is tracking those attributes in a hard-coded way. And very specific to events.

We can consider adding a more generic mechanism that tracks which realm attributes (regardless of the capability such as events) requires a different RBAC but still ... kinda adding more complexity to something already complex (RBAC).

But yeah, other than I don't think we have any other easy option.

**Comment by ssilvert** [comment]
@pedroigor @mposolda I vote for #1.  I'd rather see us go for a clean solution.  It's a little more work but it sets a precedence for the whole code base.
